src/Gui/PreferencePages/DlgSettingsEditor.cpp

```diff
@@ -163,9 +163,10 @@ DlgSettingsEditor::DlgSettingsEditor( QWidget* parent )
     QStringList labels; labels << tr("Items");
     ui->displayItems->setHeaderLabels(labels);
     ui->displayItems->header()->hide();
-    for (QVector<QPair<QString, unsigned int> >::Iterator it = d->colormap.begin(); it != d->colormap.end(); ++it) {
+    for (const auto &it: d->colormap)
+    {
         auto item = new QTreeWidgetItem(ui->displayItems);
-        item->setText(0, tr((*it).first.toLatin1()));
+        item->setText(0, tr(it.first.toLatin1()));
```

Comment:
Consider using a structured binding to unpack these pairs into more sensible variable names, e.g.
```

    for (const auto &[name, index] : d->colormap) {
        auto item = new QTreeWidgetItem(ui->displayItems);
        item->setText(0, tr(name.toLatin1()));
    }
```

Reply:
Thanks for the suggestion!

I thought that this kind of unpacking was applicable only for Python code but for sure tomorrow I'll give it a try!